shared/selene/data/metric/repository/account_activity.py

```diff
@@ -0,0 +1,122 @@
+        """Increment the deleted accounts metric on the account activity table."""
+        request = self._build_db_request(sql_file_name="increment_members_added.sql")
+        self._update_account_activity(request)
+
+    def increment_members_expired(self):
+        """Increment the deleted accounts metric on the account activity table."""
+        request = self._build_db_request(sql_file_name="increment_members_expired.sql")
+        self._update_account_activity(request)
+
+    def increment_open_dataset_added(self):
+        """Increment the deleted accounts metric on the account activity table."""
+        request = self._build_db_request(
+            sql_file_name="increment_open_dataset_added.sql"
+        )
+        self._update_account_activity(request)
+
+    def increment_open_dataset_deleted(self):
+        """Increment the deleted accounts metric on the account activity table."""
+        request = self._build_db_request(
+            sql_file_name="increment_open_dataset_deleted.sql"
+        )
+        self._update_account_activity(request)
+
+    def increment_activity(self, account: Account):
+        """Increment the activity counters depending on type of account."""
+        first_activity_of_day = (
+            account.last_activity is None
+            or account.last_activity.date() < datetime.utcnow().date()
+        )
+        if first_activity_of_day:
+            member_increment = 1 if account.membership is not None else 0
+            agreements = [agreement.type for agreement in account.agreements]
+            open_dataset_increment = 1 if OPEN_DATASET in agreements else 0
+            request = self._build_db_request(
+                sql_file_name="increment_activity.sql",
+                args=dict(
+                    member_increment=member_increment,
+                    open_dataset_increment=open_dataset_increment,
+                ),
+            )
+            self._update_account_activity(request)
+
+    def _update_account_activity(self, update_request):
+        """Update today's account activity, adding a row if it doesn't exist."""
+        row_updated = self.cursor.update(update_request)
+        if not row_updated:
+            self._add_account_activity_row()
+            self.cursor.update(update_request)
+
+    def _add_account_activity_row(self):
+        """Adds a row to the account activity table for a day that does not exist."""
+        request = self._build_db_request(sql_file_name="add_account_activity.sql")
+        self.cursor.insert(request)
+
+    def get_activity_by_date(self, activity_date: date) -> AccountActivity:
+        """Returns the account activity metrics for the given date."""
+        return self._select_one_into_dataclass(
+            dataclass=AccountActivity,
+            sql_file_name="get_account_activity_by_date.sql",
+            args=dict(activity_date=activity_date),
+        )
+
+    def delete_activity_by_date(self, activity_date: date):
```

Comment:
If it shouldn't be used why not prepend a `_` so it's not listed as a public method?